### PR TITLE
[k8s-configuration] Release v1.4.0 with S3 Bucket as Source support

### PR DESCRIPTION
**Pull request by jonathan-innis**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [ ] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [ ] Have you run `python scripts/ci/test_index.py -q` locally?

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your PR is merged into master branch, a new PR will be created to update `src/index.json` automatically.  
The precondition is to put your code inside this repo and upgrade the version in the PR but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
k8s-configuration

**Review by wangzelin007 — COMMENTED** [review]
May I ask you to add some tests for this new release?

File: src/k8s-configuration/azext_k8s_configuration/providers/FluxConfigurationProvider.py [hunk 1]
```diff
@@ -17,515 +23,1031 @@
...
+        "/subscriptions/{0}/resourceGroups/{1}/providers/{2}/{3}/{4}".format(
+            subscription_id, resource_group_name, cluster_rp, cluster_type, cluster_name
+        )
+    )
+
+    try:
+        resource = resources.get_by_id(cluster_resource_id, parent_api_version)
+        location = str(resource.location.lower())
+    except HttpResponseError as ex:
+        raise ex
```
Comment (wangzelin007) [inline src/k8s-configuration/azext_k8s_configuration/providers/FluxConfigurationProvider.py:767]: Could you please use a specific error ?

File: src/k8s-configuration/azext_k8s_configuration/providers/SourceControlConfigurationProvider.py [hunk 2]
```diff
@@ -7,124 +7,175 @@
...
+                    "(SourceControlConfigurationNotFound) The Resource {0}/{1}/{2}/"
+                    "Microsoft.KubernetesConfiguration/sourceControlConfigurations/{3}"
+                    "could not be found!".format(
+                        cluster_rp, cluster_type, cluster_name, name
+                    )
+                )
+            else:
+                message = ex.message
+            raise ResourceNotFoundError(message) from ex
+        raise ex
```
Comment (wangzelin007) [inline src/k8s-configuration/azext_k8s_configuration/providers/SourceControlConfigurationProvider.py:64]: Could you please use a specific error ?

File: src/k8s-configuration/azext_k8s_configuration/providers/FluxConfigurationProvider.py (hunk 1 above)
Comment (jonathan-innis) [reply src/k8s-configuration/azext_k8s_configuration/providers/FluxConfigurationProvider.py:767]: How would you like this to be more specific? We are already checking for one error type. If we fail for a different error reason, only then do we re-throw

File: src/k8s-configuration/azext_k8s_configuration/providers/SourceControlConfigurationProvider.py (hunk 2 above)
Comment (jonathan-innis) [reply src/k8s-configuration/azext_k8s_configuration/providers/SourceControlConfigurationProvider.py:64]: Same comment as above

**Comment by jonathan-innis** [comment]
> May I ask you to add some tests for this new release?

Hi @wangzelin007. We have E2E tests that we run in our fork that are not merged into upstream. We are unable to do unit testing because our extension resource relies on pre-created resources in Azure (i.e. a connectedCluster resource)

**Review by wangzelin007 — APPROVED** [review]

**Review by zhoxing-ms — APPROVED** [review]

**Pull request merged**